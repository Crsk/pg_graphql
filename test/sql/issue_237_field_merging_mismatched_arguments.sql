begin;
    -- https://github.com/supabase/pg_graphql/issues/237
    create table blog_post(
        id int primary key,
        a text,
        b text,
        c text,
        d text,
        e text,
        f text
    );
    insert into public.blog_post
    values (1, 'a', 'b', 'c', 'd', 'e', 'f');
    select jsonb_pretty(
      graphql.resolve($$
        query {
          blogPostCollection(filter: {
            id: { eq: 1 }
          }) {
            edges {
              node {
                a
              }
            }
          }
          blogPostCollection {
            edges {
              node {
                b
              }
            }
          }
        }
      $$)
    );
rollback;
