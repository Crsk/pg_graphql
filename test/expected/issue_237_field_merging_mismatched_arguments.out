begin;
    -- https://github.com/supabase/pg_graphql/issues/237
    create table blog_post(
        id int primary key,
        a text,
        b text,
        c text,
        d text,
        e text,
        f text
    );
    insert into public.blog_post
    values (1, 'a', 'b', 'c', 'd', 'e', 'f');
    select jsonb_pretty(
      graphql.resolve($$
        query {
          blogPostCollection(filter: {
            id: { eq: 1 }
          }) {
            edges {
              node {
                a
              }
            }
          }
          blogPostCollection {
            edges {
              node {
                b
              }
            }
          }
        }
      $$)
    );
                                             jsonb_pretty                                              
-------------------------------------------------------------------------------------------------------
 {                                                                                                    +
     "errors": [                                                                                      +
         {                                                                                            +
             "message": "Fields \"blogPostCollection\" conflict because they have differing arguments"+
         }                                                                                            +
     ]                                                                                                +
 }
(1 row)

rollback;
